bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationLocal.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.argoclima.internal.configuration;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.argoclima.internal.ArgoClimaBindingConstants;
+import org.openhab.binding.argoclima.internal.exception.ArgoConfigurationException;
+
+/**
+ * The {@link ArgoClimaConfigurationLocal} class extends base configuration parameters with ones specific
+ * to local connection (including a remote API stub / proxy)
+ *
+ * @author Mateusz Bronk - Initial contribution
+ */
+@NonNullByDefault
+public class ArgoClimaConfigurationLocal extends ArgoClimaConfigurationBase {
+    public static enum ConnectionMode {
+        LOCAL_CONNECTION,
+        REMOTE_API_STUB,
+        REMOTE_API_PROXY
+    }
+
+    public static enum DeviceSidePasswordDisplayMode {
+        NEVER,
+        MASKED,
+        CLEARTEXT
+    }
+
+    /**
+     * Argo configuration parameters specific to local connection & API stub
+     * These names are defined in thing-types.xml and get injected on instantiation
+     * through {@link org.openhab.core.thing.binding.BaseThingHandler#getConfigAs getConfigAs}
+     */
+    private String hostname = "";
+    private String localDeviceIP = "";
+    private int localDevicePort = 1001;
+    private ConnectionMode connectionMode = ConnectionMode.LOCAL_CONNECTION;
+    private boolean useLocalConnection = true;
+    private int stubServerPort = -1;
```

Comment:
Shouldn't this be 8239

Reply:
No, this is the port the HVAC listens on (they).
8239 is the default port for the stub server (us).

Reply:
Maybe `hvacListenPort` is a better name then? Is there a default port for the HVAC?

Reply:
sure, can rename to `hvacListenPort` (note though it is a full-blown change, incl. thing description and readme, so **please reconfirm** before I go about doing it).
 `1001` is "it" (hard-coded in Argo firmware). Only reason I'm even allowing to specify the port is, in case the HVAC is behind NAT, there may be some re-mapping